pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -2004,6 +2006,230 @@ func ensurePVCCreated(t *testing.T, clients test.Clients, name, namespace string
+		tb.TaskRunNamespace("foo"),
+		tb.TaskRunOwnerReference("PipelineRun", "test-pipeline-run-different-service-accs",
+			tb.OwnerReferenceAPIVersion("tekton.dev/v1beta1"),
+			tb.Controller, tb.BlockOwnerDeletion,
+		),
+		tb.TaskRunLabel("tekton.dev/pipeline", "test-pipeline"),
+		tb.TaskRunLabel("tekton.dev/pipelineRun", "test-pipeline-run-different-service-accs"),
+		tb.TaskRunLabel("tekton.dev/pipelineTask", "b-task"),
+		tb.TaskRunSpec(
+			tb.TaskRunTaskRef("b-task"),
+			tb.TaskRunServiceAccountName("test-sa-0"),
+		),
+	)
+	// Check that the expected TaskRun was created
+	actual, err := clients.Pipeline.TektonV1beta1().TaskRuns("foo").List(metav1.ListOptions{
+		LabelSelector: "tekton.dev/pipelineTask=b-task,tekton.dev/pipelineRun=test-pipeline-run-different-service-accs",
+		Limit:         1,
+	})
+
+	if err != nil {
+		t.Fatalf("Failure to list TaskRun's %s", err)
+	}
+	if len(actual.Items) != 1 {
+		t.Fatalf("Expected 1 TaskRuns got %d", len(actual.Items))
+	}
+	actualTaskRun := actual.Items[0]
+	if d := cmp.Diff(&actualTaskRun, expectedTaskRun, ignoreResourceVersion); d != "" {
+		t.Errorf("expected to see TaskRun %v created. Diff %s", expectedTaskRunName, diff.PrintWantGot(d))
+	}
+}
+
+func TestGetSkippedTasks(t *testing.T) {
+	pr := tb.PipelineRun("test-pipeline-run", tb.PipelineRunNamespace("foo"), tb.PipelineRunSpec("test-pipeline"))
+	pipelineTask := v1beta1.PipelineTask{
+		Name:    "unit-test-1",
+		TaskRef: &v1beta1.TaskRef{Name: "unit-test-task"},
+		WhenExpressions: []v1beta1.WhenExpression{{
+			Input:    "foo",
+			Operator: selection.NotIn,
+			Values:   []string{"foo", "bar"},
+		}},
+	}
+	task := tb.Task("unit-test-task", tb.TaskSpec(
+		tb.TaskResources(tb.TaskResourcesInput("workspace", resourcev1alpha1.PipelineResourceTypeGit)),
+	), tb.TaskNamespace("foo"))
+
+	expectedSkippedTasks := []v1beta1.SkippedTask{{
+		Name: "unit-test-1",
+	}}
+
+	expectedTaskRunsStatus := make(map[string]*v1beta1.PipelineRunTaskRunStatus)
+
+	expectedPipelineRunStatus := v1beta1.PipelineRunStatus{
+		PipelineRunStatusFields: v1beta1.PipelineRunStatusFields{
+			TaskRuns: expectedTaskRunsStatus,
+		},
+	}
+
+	pipelineRunState := resources.PipelineRunState{{
+		PipelineTask: &pipelineTask,
+		TaskRunName:  "test-pipeline-run-success-unit-test-1",
+		TaskRun:      nil,
+		ResolvedTaskResources: &taskrunresources.ResolvedTaskResources{
+			TaskSpec: &task.Spec,
+		},
+	}}
+	pr.Status.InitializeConditions()
+	taskrunstatus := getTaskRunsStatus(pr, pipelineRunState)
+	if d := cmp.Diff(expectedPipelineRunStatus.TaskRuns, taskrunstatus, ignoreLastTransitionTime); d != "" {
+		t.Fatalf("Expected PipelineRun status to match TaskRun(s) status, but got a mismatch: %s", diff.PrintWantGot(d))
+	}
+	dag, err := dagFromState(pipelineRunState)
+	if err != nil {
+		t.Fatalf("Could not get a dag from the state %#v: %v", pipelineRunState, err)
+	}
+	skippedtasks := getSkippedTasks(pr, pipelineRunState, dag)
```

Comment:
NIT: I would recommend checking list of skipped tasks in the pipeline run status rather than directly calling `getSkippedTasks` function here. That way, we are making sure that the status has a list of expected skipped tasks.